qemu/qmp.go

```diff
@@ -1136,3 +1176,23 @@ func (q *QMP) ExecuteVirtSerialPortAdd(ctx context.Context, id, name, chardev st
 
 	return q.executeCommand(ctx, "device_add", args, nil)
 }
+
+// ExecuteQueryMigration queries migration progress.
+func (q *QMP) ExecuteQueryMigration(ctx context.Context) (MigrationStatus, error) {
+	response, err := q.executeCommandWithResponse(ctx, "query-migrate", nil, nil, nil)
+	if err != nil {
+		return MigrationStatus{}, err
+	}
+
+	data, err := json.Marshal(response)
```

Comment:
Why is the marshalling here needed since you unmarshal it right after ?

Reply:
I think this is okay and is method used elsewhere in govmm to populate structures with the values returned from QMP calls.  The type of response is probably a map[string]interface{}.  The easiest way to get its data into a MigrationStatus is to marshal it back out to json and then unmarshal it into the destination structure.  If this were not done we'd need to write custom reflection code for each structure, which would be a pain.  There's probably a better way of doing this, but from the point of view of this PR, its fine as this is the strategy govmm currently uses to handle this problem.

Reply:
Thanks for the explanation, looks good!